components/public-api/gitpod/v1/workspace.proto

```diff
@@ -157,6 +172,59 @@ message StartWorkspaceResponse {
   Workspace workspace = 1;
 }
 
+message StopWorkspaceRequest {
+  // workspace_id specifies which workspace should be stopped.
+  //
+  // +required
+  string workspace_id = 1;
+}
+
+message StopWorkspaceResponse {}
+
+message DeleteWorkspaceRequest {
+  // workspace_id specifies the workspace that is going to delete.
+  //
+  // +required
+  string workspace_id = 1;
+}
+
+message DeleteWorkspaceResponse {}
+
+message UpdateWorkspaceRequest {
+  // workspace_id specifies the workspace to update.
+  //
+  // +required
+  string workspace_id = 1;
+
+  optional string name = 2;
+
+  // Admission describes who can access a workspace instance and its ports.
+  optional AdmissionLevel admission = 3;
+
+  optional bool pinned = 4;
+
+  // timeout is the timeout duration to stop the workspace since the last
+  // activity.
+  optional google.protobuf.Duration timeout = 5;
+
+  // git_status details the Git working copy status of the workspace.
+  optional WorkspaceGitStatus git_status = 6;
```

Comment:
I'm not sure here.

We have basically properties related to a workspace itself like pinned, admission and so on.

Then we have options related rather to a workspace status like timeout, git_status.

The difference here that updating related to workspace status will trigger reconciliation loop and fire an event via `WatchWorkspaceStatus`?. But udpating workspace properties won't make this effect.


Shoudl we rather have `UpdateWorkspace` and `UdpateWorkspaceStatus` to be explicit?

cc @csweichel @svenefftinge @geropl

Reply:
Another option unify and fire event always, but it is bigger server change which will affect dashboard as well. It feels like out of scope of migration.

Reply:
Yes, makes sense, because updating the status is something only the cluster (resp. the running workspace) should do from outside, while other properties auch as pinned are updated by users.